Common/src/geometry/CGeometry.cpp

```diff
@@ -2065,82 +2065,75 @@ void CGeometry::ComputeAirfoil_Section(su2double *Plane_P0, su2double *Plane_Nor
 
         }
 
-        for (iNode = 0; iNode < bound[iMarker][iElem]->GetnNodes(); iNode++) {
+        for (unsigned short iFace = 0; iFace < bound[iMarker][iElem]->GetnFaces(); iFace++){
+          iNode = bound[iMarker][iElem]->GetFaces(iFace,0);
+          jNode = bound[iMarker][iElem]->GetFaces(iFace,1);
           iPoint = bound[iMarker][iElem]->GetNode(iNode);
+          jPoint = bound[iMarker][iElem]->GetNode(jNode);
 
-          for (jNode = 0; jNode < bound[iMarker][iElem]->GetnNodes(); jNode++) {
-            jPoint = bound[iMarker][iElem]->GetNode(jNode);
-
-            /*--- CrossProduct concept is delicated because it allows triangles where only one side is divided by a plane.
-             that is going against the concept that all the triangles are divided twice  and causes probelms because
-             Xcoord_Index0.size() > Xcoord_Index1.size()! ---*/
-
-            if ((jPoint > iPoint) && (CrossProduct >= 0.0)
-                && ((AveXCoord > MinXCoord) && (AveXCoord < MaxXCoord))
-                && ((AveYCoord > MinYCoord) && (AveYCoord < MaxYCoord))
-                && ((AveZCoord > MinZCoord) && (AveZCoord < MaxZCoord))) {
+          if ((CrossProduct >= 0.0)
+              && ((AveXCoord > MinXCoord) && (AveXCoord < MaxXCoord))
+              && ((AveYCoord > MinYCoord) && (AveYCoord < MaxYCoord))
+              && ((AveZCoord > MinZCoord) && (AveZCoord < MaxZCoord))) {
```

Comment:
This looks messy, but the only real change in this PR is here. Instead of having an iNode and jNode loop that traverse the nodes of each element, there is one iFace loop that traverses the edges of the elements and sets iNode and jNode from that face. The rest of this routine is unchanged.

Previously for quads with iNode and jNode loops, it would try and find intersections along edges that did not exist (imagine a diagonal cut of a quad). This would then mess up the routine further down.

Reply:
The lines in question are above the code snippet linked